### PR TITLE
WIP Incident Redis Refactoring Madness

### PR DESCRIPTION
**Pull request by captncraig**
- [x] alerts run
- [x] dashboard renders
- [x] unknowns trigger
- [x] incident view working
- [ ] alert counts reporting
- [x] restore historical incidents
- [x] action notifications working again
- [x] tests passing
- [x] dependency function working.

### CODE REVIEW

File: cmd/bosun/sched/check.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package sched
 
 import (
 	"fmt"
+	"log"
```
Comment (kylebrandt) [inline cmd/bosun/sched/check.go]: should be using slog

**Comment by kylebrandt** [comment]
Bug: Dashboard shows alert key now instead of subject:

Master:
image

Branch:
image

**Comment by kylebrandt** [comment]
Bug: Full History links on dashboard don't work anymore:
Link:
image


View:
image

**Comment by kylebrandt** [comment]
Bug: Dashboard alert expands that show the template body no longer work:

Master:
image

Branch:
image

**Comment by kylebrandt** [comment]
Bug: Silence doesn't show alert will be silenced when using test on the silence page:

Master:
image


Branch:
image

**Comment by kylebrandt** [comment]
Bug? Stuff still gets written to bold when stopping bosun:

```2016/01/08 10:19:15 info: bolt.go:76: wrote notifications: 48.00B
2016/01/08 10:19:15 info: bolt.go:100: save to db complete```

**Comment by kylebrandt** [comment]
Bug? Migrating state runs on a second run, not just the first according to the log:

```2016/01/08 10:21:21 info: bolt.go:487: migrating state```

**Comment by kylebrandt** [comment]
Bug: New template erros on many alerts that we don't see on prod:

```2016/01/08 10:22:08 info: check.go:240: hardware{host=nyhq-fs01}: template: hardware:1:65: executing "hardware" at <.Group.host>: Group is not a field of struct type *sched.Context
2016/01/08 10:22:08 info: check.go:253: hardware{host=nyhq-fs01}: template: header:10:19: executing "header" at <.Group.host>: Group is not a field of struct type *sched.Context
2016/01/08 10:22:08 info: check.go:266: hardware{host=nyhq-fs01}: template: header:10:19: executing "header" at <.Group.host>: Group is not a field of struct type *sched.Context
2016/01/08 10:22:08 info: check.go:279: hardware{host=nyhq-fs01}: template: hardware:1:65: executing "hardware" at <.Group.host>: Group is not a field of struct type *sched.Context```

**Comment by kylebrandt** [comment]
Bug: Historical testing doesn't seem to work:

Master:
image

Branch:
image

**Comment by captncraig** [comment]
Multiple migrations fixed. May need to flush redis to avoid duplicate data.

**Comment by kylebrandt** [comment]
```2016/01/19 09:44:27 info: check.go:256: hardware{host=ny-sql04}: template: header:22:48: executing "header" at <.Last.IncidentId>: IncidentId is not a field of struct type models.Event
2016/01/19 09:44:27 info: check.go:269: hardware{host=ny-sql04}: template: header:22:48: executing "header" at <.Last.IncidentId>: IncidentId is not a field of struct type models.Event```

**Comment by kylebrandt** [comment]
image

**Comment by kylebrandt** [comment]
image

**Pull request merged**